### PR TITLE
Remove Android Pre-AppCompat

### LINKED ISSUE #20: [Spec] Remove Android Pre-AppCompat
# Description

Remove pre-AppCompat renderers and supporting classes from the Android Platform.

# API Changes

| Class to Remove | AppCompat Replacement |
| ------------- | ------------- |
| FormsApplicationActivity | FormsAppCompatActivity |
| Platform | AppCompat.Platform |
| ButtonRenderer | AppCompat.ButtonRenderer |
| CarouselPageRenderer | AppCompat.CarouselPageRenderer |
| FrameRenderer | AppCompat.FrameRenderer |
| MasterDetailRenderer | AppCompat.MasterDetailPageRenderer |
| NavigationRenderer | AppCompat.NavigationPageRenderer |
| PickerRenderer | AppCompat.PickerRenderer |
| SwitchRenderer | AppCompat.SwitchRenderer |
| TabbedRenderer | AppCompat.TabbedPageRenderer |

# Backward Compatibility

This will be a breaking change for anyone still using the pre-AppCompat renderers. Those users will need to update their applications to use FormsAppCompatActivity and update any renderer customization to use the AppCompat renderers.

# Difficulty : Low

This mostly entails deleting files.

### PR DESCRIPTION
**Pull request by domagojmedo**
### Description of Change ###

Removed Android Pre-AppCompat

### Issues Resolved ### 

- fixes #20 

### API Changes ###

Added:
    Moved methods and properties from Platform to AppCompat.Platform:
-  internal const string AppCompat.Platform.CloseContextActionsSignalName 
-  internal static readonly BindableProperty AppCompat.Platform.RendererProperty
-  public static void ClearRenderer(AView renderedView)
-  public static IVisualElementRenderer CreateRenderer(VisualElement element)
-  internal static IVisualElementRenderer CreateRenderer(VisualElement element, Context context)
-  internal static IVisualElementRenderer CreateRenderer(VisualElement element, FragmentManager fragmentManager, Context context)
-  public static IVisualElementRenderer CreateRendererWithContext(VisualElement element, Context context)
-  public static IVisualElementRenderer GetRenderer(VisualElement bindable)
-  public static void SetRenderer(VisualElement bindable, IVisualElementRenderer value)
-  internal ViewGroup GetViewGroup()
-  internal static readonly BindableProperty PageContextProperty 
-  internal static void SetPageContext(BindableObject bindable, Context context)
-  static Context GetPreviewerContext(Element element)
 
 Removed clases:
 - System.Maui.ControlGallery.Android.FormsApplicationActivity
 - System.Maui.Platform.Android.FormsApplicationActivity
 - System.Maui.Platform.Android.Platform
 - System.Maui.Platform.Android.ButtonRenderer
 - System.Maui.Platform.Android.CarouselPageRenderer
 - System.Maui.Platform.Android.FrameRenderer
 - System.Maui.Platform.Android.MasterDetailRenderer
 - System.Maui.Platform.Android.NavigationRenderer
 - System.Maui.Platform.Android.PickerRenderer
 - System.Maui.Platform.Android.NavigationRenderer
 - System.Maui.Platform.Android.SwitchRenderer
 - System.Maui.Platform.Android.TabbedRenderer
 
### Platforms Affected ### 

- Android

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=174) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=174) before we can accept your contribution.<br/><br/>:x: domagojmedo [sign now](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=174)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/maui?pullRequest=174) it.</sub>

File: System.Maui.ControlGallery.Android/Tests/TestingPlatformService.cs [hunk 1]
```diff
@@ -11,9 +10,11 @@ class TestingPlatformService : ITestingPlatformService
 	{
 		public async Task CreateRenderer(VisualElement visualElement)
 		{
-			await Device.InvokeOnMainThreadAsync(() => 
-				Platform.Android.Platform.CreateRendererWithContext(visualElement,
-					DependencyService.Resolve<Context>()));
+			//await Device.InvokeOnMainThreadAsync(() =>
+			//	Platform.Android.AppCompat.Platform.CreateRendererWithContext(visualElement,
```
Comment (domagojmedo) [inline System.Maui.ControlGallery.Android/Tests/TestingPlatformService.cs:14]: Pre-AppCompat Platform was public, now it's internal, not sure how to resolve this

File: System.Maui.ControlGallery.Android/RegistrarValidationService.cs [hunk 2]
```diff
@@ -21,15 +21,15 @@ public bool Validate(VisualElement element, out string message)
 			if (element == null)
 				return true;
 
-			var renderer = Platform.Android.Platform.CreateRendererWithContext(element, _context);
+			//var renderer = Platform.Android.AppCompat.Platform.CreateRendererWithContext(element, _context);
```
Comment (domagojmedo) [inline System.Maui.ControlGallery.Android/RegistrarValidationService.cs:24]: Pre-AppCompat Platform was public, now it's internal, not sure how to resolve this

**Comment by rmarinho** [comment]
@domagojmedo  thanks very muck , right now we will start this work on the Xamarin.Forms repo on the main branch.
Do you think you can do this work there?

**Comment by domagojmedo** [comment]
Yes, I'll try to do it by the end of the weekend

**Comment by PureWeen** [comment]
Thank you @domagojmedo  !!

going to close this PR here for now

**Pull request closed without merging**